### PR TITLE
Add Quiet and Silent switches

### PR DESCRIPTION
**Pull request by tbateson**
- Added `-Quiet` and `-Silent` switches to suppress output; `-Quiet` suppresses `stdout`, `-Silent` suppresses `stderr` and `stdout`
    - `-Silent` only covers errors reported by `pwr.ps1`, not sure about errors reported by other executables
- Partially fixed `-Offline`, it was still fetching even when specified; I fixed this by moving the handling of offline to earlier in the processing
- Fixed variable style from `$var` -> `$Var`
- Removed `PwrWebPath` environment variable usage

### CODE REVIEW

File: src/pwr.ps1 [hunk 1]
```diff
@@ -87,81 +93,101 @@ Class SemanticVersion : System.IComparable {
...
+		Write-PwrHost " $($ExecutionContext.SessionState.Path.CurrentLocation)$('>' * ($NestedPromptLevel + 1))" -NoNewline
 		return ' '
 	} else {
-		"PS $($executionContext.SessionState.Path.CurrentLocation)$('>' * ($nestedPromptLevel + 1)) "
+		"PS $($ExecutionContext.SessionState.Path.CurrentLocation)$('>' * ($NestedPromptLevel + 1)) "
 	}
 }
 
-function Invoke-PwrWebRequest($Uri, $Headers, $OutFile, [switch]$UseBasicParsing) {
-	if ($Offline) {
```
Comment (troyaf) [inline src/pwr.ps1:144]: This was here for good reason. Without this you need to always know where web requests happen and have offline blocks for it. Case in point: `pwr update` now succeeds with `-offline` specified.

**Review by troyaf — CHANGES_REQUESTED** [review]

File: src/pwr.ps1 (hunk 1 above)
Comment (tbateson) [reply src/pwr.ps1:144]: Then we need to do that. As it was, this function would just use `-Offline` to spit out this error message, then continue executing the web request. This is not what we want, when you specify `-Offline` it should never try to hit the network.

File: src/pwr.ps1 [hunk 2]
```diff
@@ -87,81 +93,101 @@ Class SemanticVersion : System.IComparable {
...
-		Write-Error 'pwr: web request while running offline'
+function Write-PwrOutput {
+	if (!($Quiet) -and !($Silent)) {
```
[the remark this answers is not in the document]
Comment (tbateson) [reply src/pwr.ps1:150]: I can make these changes, but there are other places where you're using `if (!($BooleanVar))`

[the remark this answers is not in the document]
Comment (troyaf) [reply src/pwr.ps1:150]: Consistency is hard.

File: src/pwr.ps1 (hunk 1 above)
Comment (troyaf) [reply src/pwr.ps1:144]: I'll test this on main, but it shouldn't have. `ErrorActionProgress` is `Stop` meaning the write-error alone should have stopped execution.

Comment (tbateson) [reply src/pwr.ps1:144]: It doesn't return after this error message, and `$Offline` wasn't gating anything from accessing this function previously.

Comment (troyaf) [reply src/pwr.ps1:144]: It did not continue on `main`. I'd be curious to know how you produced that behavior.

Comment (troyaf) [reply src/pwr.ps1:144]: Granted, the console was ugly, but the desired behavior was there. Agree that the feedback to the user needed improvement.

File: src/pwr.ps1 [hunk 3]
```diff
@@ -376,54 +404,60 @@ function Get-PwrPackages {
...
 }
 
-function Invoke-PwrPackagePull($pkg) {
-	if (Test-PwrPackage $pkg) {
-		Write-Output "pwr: $($pkg.ref) already exists"
+function Invoke-PwrPackagePull($Pkg) {
+	if (Test-PwrPackage $Pkg) {
+		Write-PwrOutput "pwr: $($Pkg.ref) already exists"
+	} elseif ($Offline) {
+		Write-PwrOutput "pwr: skipping fetch $($Pkg.ref) while running offline"
```
[the remark this answers is not in the document]
Comment (tbateson) [reply src/pwr.ps1]: Should we just call `exit` then after writing the message?

[the remark this answers is not in the document]
Comment (troyaf) [reply src/pwr.ps1]: We can make this an error log, and update the error log to exit 1.

File: src/pwr.ps1 (hunk 1 above)
Comment (tbateson) [reply src/pwr.ps1:144]: Maybe I'm missing, does `Write-Error` throw an exception?

File: src/pwr.ps1 (hunk 3 above)
[the remark this answers is not in the document]
Comment (tbateson) [reply src/pwr.ps1]: I've replaced the `Write-Error` with `Write-Host -ForegroundColor Red`. From a usability perspective I don't see much the user can do with the stacktrace.

File: src/pwr.ps1 (hunk 1 above)
Comment (tbateson) [reply src/pwr.ps1:144]: Looks like it's equivalent to `throw` with more features.

File: src/pwr.ps1 (hunk 3 above)
[the remark this answers is not in the document]
Comment (troyaf) [reply src/pwr.ps1]: Sounds good

File: src/pwr.ps1 (hunk 1 above)
Comment (troyaf) [reply src/pwr.ps1:144]: Yeah, it's basically a throw, but I set the pwr script to immediately stop execution when it sees an error. Our new write-fatal is close to equivalent

File: src/pwr.ps1 (hunk 3 above)
[the remark this answers is not in the document]
Comment (troyaf) [reply src/pwr.ps1]: Actually, I think there were a few places that I was catching an error so a straight exit might not be exactly what I want. I'll go check the uses

**Pull request merged**